master/src/ComputerZoomWidget.cpp

```diff
@@ -0,0 +1,86 @@
+/*
+ * ComputerZoomWidget.cpp - fullscreen preview widget
+ *
+ * Copyright (c) 2006-2020 Tobias Junghans <tobydox@veyon.io>
+ *
+ * This file is part of Veyon - https://veyon.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QIcon>
+#include <QLayout>
+
+#include "ComputerControlInterface.h"
+#include "ComputerZoomWidget.h"
+#include "VeyonConfiguration.h"
+#include "VeyonMasterInterface.h"
+#include "PlatformCoreFunctions.h"
+#include "VncViewWidget.h"
+
+
+ComputerZoomWidget::ComputerZoomWidget(const ComputerControlInterface::Pointer& computerControlInterface ) :
+	QWidget( nullptr ),
+	m_computerControlInterface( computerControlInterface ),
+	m_vncView( new VncViewWidget( computerControlInterface->computer().hostAddress(), -1, this, VncView::RemoteControlMode ) )
```

Comment:
FYI: I'll later add an overload to `VncViewWidget` which takes a `ComputerControlInterface` so the existing connection can be reused similar to how it's implemented in `SpotlightPanel`.

Reply:
That sounds extremely cool, as it would probably remove the delay due to opening a new connection. I wish I my skills would enable me to do this for you, but I'll look at what you will do and try to learn how it is done.